src/main/scala/org/dia/core/SciDataset.scala

```diff
@@ -111,6 +127,35 @@ class SciDataset(val variables: mutable.HashMap[String, Variable],
    */
   def apply(key: String): Variable = variables(key)
 
+
+  /**
+   * Updates the SciDataset by inserting a new Variable
+   * for the given key.
+   *
+   * Usage sciD("key") = variable
+   * @param key the variable key to rename to
+   * @param variable the variable
+   * @return the modifed SciDataset
+   */
+  def update(key: String, variable: Variable): SciDataset = {
+    variables(key) = variable
+    this
+  }
+
+  /**
+   * Updates the SciDataset by inserting a new attribute
+   * for the given key.
+   *
+   * Usage sciD("key") = attributeString
+   * @param key the variable key to rename to
+   * @param attribute the attribute name
+   * @return the modified SciDataset
+   */
+  def update(key: String, attribute: String): SciDataset = {
+    attributes(key) = attribute
+    this
+  }
+
```

Comment:
For those it concerns the update function in scala is how x("s") = y is done.
It's the same case with hashmaps.
The last parameter goes on the right side of the equal sign.
The first parameter until the last parameter goes on the left side of the equal sign.

For more info visit here :
http://daily-scala.blogspot.com/2009/08/apply-update-methods.html